src/kv/RocksDBStore.cc

```diff
@@ -1421,21 +1421,32 @@ void RocksDBStore::get_statistics(Formatter *f)
 	     << dendl;
     return;
   }
-
+  
   if (cct->_conf->rocksdb_collect_compaction_stats) {
-    std::string stat_str;
-    bool status = db->GetProperty("rocksdb.stats", &stat_str);
-    if (status) {
-      f->open_object_section("rocksdb_statistics");
+    vector<rocksdb::ColumnFamilyHandle*> handles;
+    handles.push_back(default_cf);
+    for (auto cf : cf_handles) {
+      for (auto shard_cf : cf.second.handles) {
+        handles.push_back(shard_cf);
+      }
+    }
+    f->open_object_section("rocksdb_statistics");
+    for (auto handle : handles) {
+      std::string stat_str;
+      bool status = db->GetProperty(handle, "rocksdb.stats", &stat_str);
+      if (!status) {
+        continue;
```

Comment:
Please log an error at this point

Reply:
added

Reply:
IMO better use derr.

Reply:
And can we get readable CF name from ColumnFamilyHandle? If so - please print it in the error message too.

Reply:
Thanks for your suggestion. Added.